### PR TITLE
add new API pruning examples

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

add new API pruning examples

### CODE REVIEW

File: examples/pytorch/image_recognition/torchvision_models/quantization/ptq/cpu/ipex/README.md [hunk 1]
```diff
@@ -50,7 +50,15 @@ refer [intel/intel-extension-for-pytorch at icx (github.com)](https://github.com
    >
    > Support IPEX version >= 1.8.0, 1.12.0 version need build from source and apply patch.
 
+<<<<<<< HEAD
```
Comment (XinyuYe-Intel) [inline examples/pytorch/image_recognition/torchvision_models/quantization/ptq/cpu/ipex/README.md:53]: conflict not solved

Comment (xin3he) [reply examples/pytorch/image_recognition/torchvision_models/quantization/ptq/cpu/ipex/README.md:53]: thx~

File: examples/pytorch/image_recognition/torchvision_models/optimization_pipeline/prune_and_ptq/eager/main.py [hunk 2]
```diff
@@ -4,103 +4,377 @@
...
-    and callable(quantize_models.__dict__[name]))
+model_names = sorted(name for name in models.__dict__
+                     if name.islower() and not name.startswith("__")
+                     and callable(models.__dict__[name]))
 
 parser = argparse.ArgumentParser(description='PyTorch ImageNet Training')
 parser.add_argument('data', metavar='DIR',
                     help='path to dataset')
-parser.add_argument('-a', '--arch', metavar='ARCH', default='resnet18',
+parser.add_argument('-t', '--topology', metavar='ARCH', default='resnet18',
```
Comment (PenghuiCheng) [inline examples/pytorch/image_recognition/torchvision_models/optimization_pipeline/prune_and_ptq/eager/main.py:24]: align with public example code

**Review by PenghuiCheng — APPROVED** [review]

**Review by PenghuiCheng — APPROVED** [review]

File: examples/pytorch/image_recognition/torchvision_models/optimization_pipeline/prune_and_ptq/eager/main.py [hunk 3]
```diff
@@ -4,103 +4,377 @@
...
+
+    if args.quantize:
+        from neural_compressor import PostTrainingQuantConfig
+        from neural_compressor import quantization
+        q_conf = PostTrainingQuantConfig()
+        q_model = quantization.fit(model.model,
+                                    q_conf,
+                                    calib_dataloader=val_loader,
+                                    eval_func=eval_func)
         q_model.save(args.tuned_checkpoint)
```
Comment (PenghuiCheng) [inline examples/pytorch/image_recognition/torchvision_models/optimization_pipeline/prune_and_ptq/eager/main.py:227]: Where is pruning?

**Review by PenghuiCheng — APPROVED** [review]

**Review by PenghuiCheng — APPROVED** [review]

**Pull request merged**